Storage/Rust/RustLogins.swift

```diff
@@ -815,75 +828,113 @@ public class RustLogins {
         keychain.removeObject(forKey: rustKeys.loginsSaltKeychainKey, withAccessibility: .afterFirstUnlock)
     }
 
-    public func getStoredKey() throws -> String {
+    private func resetLoginsAndKey(rustKeys: RustLoginEncryptionKeys,
+                                   completion: @escaping (Result<String, NSError>) -> Void) {
+        self.wipeLocalEngine().upon { result in
+            guard result.isSuccess else {
+                completion(.failure(result.failureValue! as NSError))
+                return
+            }
+
+            do {
+                let key = try rustKeys.createAndStoreKey()
+                completion(.success(key))
+            } catch let error as NSError {
+                self.logger.log("Error creating logins encryption key",
+                                level: .warning,
+                                category: .storage,
+                                description: error.localizedDescription)
+                completion(.failure(error))
+            }
+        }
+    }
+
+    public func getStoredKey(completion: @escaping (Result<String, NSError>) -> Void) {
+        guard UIApplication.shared.isProtectedDataAvailable else {
```

Comment:
OMG! This is really boneheaded. I deleted the additional commits with the `isProtectedDataAvailable` logic but forgot to remove this clause from the original commit which is TOTALLY why the tests are crashing smh. Removing this now. Sorry about that.